### PR TITLE
[DF] ROOT-9468 - Add TColumnValue extern templates for common types

### PR DESCRIPTION
**Pull request by dpiparo**
this mitigates ROOT-9468 speeding up considerably the instantiation of large templates, such as Snapshot of many columns.
It is helpful in general too.

### CODE REVIEW

**Comment by dpiparo** [comment]
@phsft-bot build

File: tree/dataframe/inc/ROOT/RDFNodes.hxx [hunk 1]
```diff
@@ -278,7 +278,7 @@ class TColumnValue {
 public:
    static constexpr bool fgMustUseRVec = MustUseRVec;
 
-   TColumnValue() = default;
+   TColumnValue() {};
```
Comment (Axel-Naumann) [inline tree/dataframe/inc/ROOT/RDFNodes.hxx]: Why is this change needed?

Comment (dpiparo) [reply tree/dataframe/inc/ROOT/RDFNodes.hxx]: If I don't do this, I get missing symbols on ubu 16 gcc5 :(
